### PR TITLE
Metasploit cheatsheet

### PR DESCRIPTION
**Pull request by rohitzidu**
What does your Instant Answer do?
cheatsheet goodies

What is the data source for your Instant Answer? (Provide a link if possible)
http://www.cheatography.com/huntereight/cheat-sheets/metasploit-4-5-0-dev-15713/

Why did you choose this data source?
It is cool cheatsheet and most people used it

Are there any other alternative (better) data sources?
yes there many data source, but it is good and descriptive compare to other.

Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)
security, sysadmin

Is this Instant Answer connected to a DuckDuckHack Instant Answer idea?
Yes, it is cheatsheet for security professional and pen tester.

Are you having any problems? Do you need our help with anything?
No

Where did you hear about DuckDuckHack? (For first time contributors)
i am co-organizer for Delhi group quack and hack & it will example for them.

What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)
sure
metasploit

-----
IA Page: https://duck.co/ia/view/metasploit_cheat_sheet

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # DuckDuckHack Goodies [![Build Status](https://travis-ci.org/duckduckgo/zeroclickinfo-goodies.png?branch=master)](https://travis-ci.org/duckduckgo/zeroclickinfo-goodies)
 
+[![Join the chat at https://gitter.im/rohitzidu/zeroclickinfo-goodies](https://badges.gitter.im/Join%20Chat.svg)](https://gitter.im/rohitzidu/zeroclickinfo-goodies?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
```
Comment (moollaza) [inline README.md]: Whoops I don't think this is supposed to be in here? The README should stay as is :smile:

File: share/goodie/cheat_sheets/metasploit.json [hunk 2]
```diff
@@ -0,0 +1,340 @@
+{
+    "id": "Metasploit_cheat_sheet",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/metasploit.json]: This actually needs to be all lowercase letters: `metasploit_cheat_sheet`

File: share/goodie/cheat_sheets/metasploit.json [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+    "name": "Metasploit",
+    "description": "Framework provides information about security vulnerabilities",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/metasploit.json]: typo? "**A** frame **that** provides ..."

File: share/goodie/cheat_sheets/metasploit.json [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+    "metadata": {
+        "sourceName": "metasploit",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/metasploit.json]: Looks like the source should really be `Cheatography`?

File: share/goodie/cheat_sheets/metasploit.json [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+        }, {
+            "val": "Switch Between Workspaces",
+            "key": "workspace"
+        }, {
+            "val": "NMAP Scan into Database",
+            "key": "db_nmap"
+        }],
+        "Core Commands": [{
+            "val": "Display Help",
+            "key": "? or help"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/metasploit.json]: These alternatives should be separate entries:

```
{
  "val": "Display Help",
  "key": "?"
},
{
  "val": "Display Help (alternative)",
  "key": "help"
}
```

**Comment by moollaza** [comment]
@rohitzidu thanks a lot for contributing, we really appreciate it! This looks pretty good so far, some minor fixes are needed. I've left some comments.

**Comment by rohitzidu** [comment]
Thanks moolaza,

For valuable comments and workaround, I will complete and check all comments; I will let you know after all alter.

Rohit

@rohitzedu

 rohitzidu

 +91 7503662776

**Comment by namandistro** [comment]
@rohitzidu I guess you should not put your personal detail in public like this.

**Comment by rohitzidu** [comment]
@moollaza Kindly update when it will release; mean while we are sending some more ideas and one by one we will complete all goodies and spice. Actually this will motivate our members and people who are keen to work and let me if i want to change anything else


@namandistro  actually sent message via my mail Id and my signature went off with it.

**Comment by namandistro** [comment]
@rohitzidu You can even it :p

File: README.md (hunk 1 above)
Comment (moollaza) [inline README.md]: @rohitzidu it looks like you used Gitter to add a badge to the repo? Gitter updated your fork of the repo with a new url. You can `git revert` the commit to get rid of this. As long as it's not part of the pull request, I'm happy.

**Comment by moollaza** [comment]
@rohitzidu once it's all fixed up and I've merged it, I need to setup an internal release. Once that's done it will go live.

**Comment by rohitzidu** [comment]
Revert Gitter READ.md file

**Comment by rohitzidu** [comment]
Status of Metasploit cheatsheet Team.

**Comment by moollaza** [comment]
@rohitzidu it looks like you've added in another Cheat Sheet here? Can you please remove it? We need PR's to only contain 1 IA at a time.

**Comment by moollaza** [comment]
@rohitzidu as well, we've made some recent changes to the location of the JSON files, I'm really sorry, but I'll need you to move the files into the new /json/ directory. Using `git mv ...` should do the trick!

File: share/goodie/cheat_sheets/metasploit.json [hunk 6]
```diff
@@ -0,0 +1,439 @@
+{
+    "id": "metasploit_cheat_sheet",
+    "name": "Metasploit",
+    "description": "A framework provides information about security vulnerabilities",
+    "metadata": {
+        "sourceName": "Cheatography",
+        "sourceUrl": "http://www.cheatography.com/huntereight/cheat-sheets/metasploit-4-5-0-dev-15713/"
+    },
+    "section_order": [
```
Comment (moollaza) [inline share/goodie/cheat_sheets/metasploit.json]: We now have templates for Cheat Sheets! It looks like this is a command line program? If so then we should add:
```json
"template_type": "terminal",
```

The result willq looks awesome!

Comment (moollaza) [inline share/goodie/cheat_sheets/metasploit.json]: Here's the result!
screen shot 2015-08-18 at 12 56 02

File: share/goodie/cheat_sheets/metasploit.json [hunk 7]
```diff
@@ -0,0 +1,439 @@
+{
+    "id": "metasploit_cheat_sheet",
+    "name": "Metasploit",
+    "description": "A framework provides information about security vulnerabilities",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/metasploit.json]: Looks like a small typo? "A framework **that** provides information about security vulnerabilities"

File: share/goodie/cheat_sheets/metasploit.json (hunk 6 above)
Comment (rohitzidu) [inline share/goodie/cheat_sheets/metasploit.json]: Really awesome!!! Cool stuff

Thanks @mollza for all information , I will do all changes today only and let you know.

Rohit

File: share/goodie/cheat_sheets/metasploit.json [hunk 8]
```diff
@@ -0,0 +1,439 @@
...
+        {
+            "val": "Display Services",
+            "key": "services"
+        },
+        {
+            "val": "Display Vulnerabilities",
+            "key": "vulns"
+        },
+        {
+            "val": "Switch Between Workspaces",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/metasploit.json]: Can we possibly switch all the `val`'s to use sentence casing? Right now they're using Title Casing, which looks a little odd.

**Comment by moollaza** [comment]
@rohitzidu I've left a little more feedback. Once the changes have been made to Metasploit it's ready for merge!

I'll take a look at Nikto now...

File: share/goodie/cheat_sheets/nikto.json [hunk 9]
```diff
@@ -0,0 +1,359 @@
+{
+    "id": "nikto_cheat_sheet",
+    "name": "Nikto",
+    "description": "A Nikto is perl based vulnerability scanner that scans webserver for thousands of vulnerabilities and other known issues",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/nikto.json]: typo? "That scans webserver**s**"? OR "That scans **your** webserver ... "?

File: share/goodie/cheat_sheets/nikto.json [hunk 10]
```diff
@@ -0,0 +1,359 @@
...
+    "metadata": {
+        "sourceName": "r00tsec",
+        "sourceName_1": "cirt",
+        "sourceUrl": "http://www.r00tsec.com/2014/06/cheatsheet-nikto-cheat-sheet.html",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/nikto.json]: only SourceName and SourceURL are allowed. Please remove the others.

File: share/goodie/cheat_sheets/nikto.json [hunk 11]
```diff
@@ -0,0 +1,359 @@
...
+        "sourceUrl_1": "https://cirt.net/nikto2-docs/options.html#id2741238"
+    },
+    "section_order": [
```
Comment (moollaza) [inline share/goodie/cheat_sheets/nikto.json]: It looks like this one can benefit from `"template_type": "terminal"` as well?

File: share/goodie/cheat_sheets/nikto.json [hunk 12]
```diff
@@ -0,0 +1,359 @@
...
+        "Tuning Options",
+        "Mutate Options",
+        "Output File Format",
+        "Basic Options",
+        "Advanced Options"
+    ],
+    "sections": {
+        "Installation": [
+            {
+                "val": "Install nikto on ubuntu",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/nikto.json]: I think `Nikto` should be capitalized throughout? Same goes for "Ubuntu"

File: share/goodie/cheat_sheets/nikto.json [hunk 13]
```diff
@@ -0,0 +1,359 @@
...
+                "key": "sudo apt-get install nikto"
+            },
+            {
+                "val": "Install nikto on windows (On windows first install the perl interpreter)",
+                "key": "C:\\pentest\\nikto-2.1.5>perl nikto.pl -h example.com"
+            },
+            {
+                "val": "Baisc Usage",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/nikto.json]: Typo: "Basic"

File: share/goodie/cheat_sheets/nikto.json (hunk 11 above)
Comment (rohitzidu) [inline share/goodie/cheat_sheets/nikto.json]: Hey Zaahir,

Pretty awesome consideration, in nikto cheat sheet I have used two source and reference web site in json file. 

Is there possibility both can come on instant answer, or I have to remove one. What you suggest?



-Rohit

Comment (moollaza) [inline share/goodie/cheat_sheets/nikto.json]: >Is there possibility both can come on instant answer, or I have to remove one. What you suggest?

Right now we only support 1. I guess you can leave the additional info in the JSON, but we won't be able to use it for now.

**Comment by rohitzidu** [comment]
@moollaza is it done
